plugins/catalog-backend-module-gitlab/src/providers/GitlabOrgDiscoveryEntityProvider.ts

```diff
@@ -0,0 +1,364 @@
+        options.schedule ??
+        options.scheduler!.createScheduledTaskRunner(providerConfig.schedule!);
+
+      providers.push(
+        new GitlabOrgDiscoveryEntityProvider({
+          ...options,
+          config: providerConfig,
+          integration,
+          taskRunner,
+        }),
+      );
+    });
+    return providers;
+  }
+
+  private constructor(options: {
+    config: GitlabProviderConfig;
+    integration: GitLabIntegration;
+    logger: Logger;
+    taskRunner: TaskRunner;
+  }) {
+    this.config = options.config;
+    this.integration = options.integration;
+    this.logger = options.logger.child({
+      target: this.getProviderName(),
+    });
+    this.scheduleFn = this.createScheduleFn(options.taskRunner);
+  }
+
+  getProviderName(): string {
+    return `GitlabOrgDiscoveryEntityProvider:${this.config.id}`;
+  }
+
+  async connect(connection: EntityProviderConnection): Promise<void> {
+    this.connection = connection;
+    await this.scheduleFn();
+  }
+
+  private createScheduleFn(taskRunner: TaskRunner): () => Promise<void> {
+    return async () => {
+      const taskId = `${this.getProviderName()}:refresh`;
+      return taskRunner.run({
+        id: taskId,
+        fn: async () => {
+          const logger = this.logger.child({
+            class: GitlabOrgDiscoveryEntityProvider.prototype.constructor.name,
+            taskId,
+            taskInstanceId: uuid.v4(),
+          });
+
+          try {
+            await this.refresh(logger);
+          } catch (error) {
+            logger.error(`${this.getProviderName()} refresh failed`, error);
+          }
+        },
+      });
+    };
+  }
+
+  private async refresh(logger: Logger): Promise<void> {
+    if (!this.connection) {
+      throw new Error(
+        `Gitlab discovery connection not initialized for ${this.getProviderName()}`,
+      );
+    }
+
+    const client = new GitLabClient({
+      config: this.integration.config,
+      logger: logger,
+    });
+
+    const users = paginated<GitLabUser>(options => client.listUsers(options), {
+      page: 1,
+      per_page: 100,
+      active: true,
+    });
+
+    const groups = paginated<GitLabGroup>(
+      options => client.listGroups(options),
+      {
+        page: 1,
+        per_page: 100,
+      },
+    );
+
+    const idMappedGroup: { [groupId: number]: GitLabGroup } = {};
+
+    const res: Result = {
+      scanned: 0,
+      matches: [],
+    };
+
+    const groupRes: GroupResult = {
+      scanned: 0,
+      matches: [],
+    };
+
+    for await (const group of groups) {
+      if (!this.config.groupPattern.test(group.full_path ?? '')) {
+        continue;
+      }
+
+      groupRes.scanned++;
+      groupRes.matches.push(group);
+
+      idMappedGroup[group.id] = group;
+    }
+
+    for await (const user of users) {
+      if (!this.config.userPattern.test(user.email ?? '')) {
+        continue;
+      }
+
+      res.scanned++;
+
+      if (user.active) {
+        continue;
+      }
```

Comment:
should this be negated as `if (!user.active)` ?

Reply:
that is fixed